examples/recipes/non-additivity/schema/UsersRefactored.js

```diff
@@ -0,0 +1,40 @@
+cube(`UsersRefactored`, {
+  extends: Users,
+
+  measures: {
+    // Replaced with an approximate distinct count
+    // When used with Postgres, requires a HypedLogLog extension:
+    // https://github.com/citusdata/postgresql-hll
+    // distinctAges: {
+    //   sql: `age`,
+    //   type: `countDistinctApprox`,
+    // },
+
+    // Decomposed into a formula with additive measures
+    avgAge: {
+      sql: `${CUBE.ageSum} / ${CUBE.count}`,
+      type: `number`,
+    },
+
+    ageSum: {
+      sql: `age`,
+      type: `sum`,
+    },
+
+    count: {
+      type: `count`,
+    },
+  },
+
+  preAggregations: {
+    main: {
+      measures: [
+        CUBE.ageSum,
```

Comment:
@igorlukanin are you sure here's `ageSum`, not `avgAge`?

Reply:
I am! `avgAge` will be calculated on-the-fly based on `ageSum` and `count`, that's the whole point 😄